app.py

```diff
@@ -204,4 +188,39 @@ def launch_publish(self, entity_type, entity_ids):
             # hook didn't know how to launch this
             # just use std associated file launch
             self.launch(path_on_disk)
-        
\ No newline at end of file
+
+    def get_path_on_disk(self, published_file):
+        """
+        Retrieve, if possible, the path on disk of a PublishedFile.
+
+        :param published_file: A PublishedFile dictionary.
+        :returns: The path on disk if found, otherwise ``None``.
+        """
+        path_on_disk = published_file.get("path", {}).get("local_path")
+        if path_on_disk:
+            return path_on_disk
+        # If this PublishedFile came from a zero config publish, it will
+        # have a file URL rather than a local path.
+        path_on_disk = published_file.get("path").get("url")
```

Comment:
Why getting it again?

Reply:
Sorry I was confused by the same variable name, we're getting the url here. May be use another variable?